lib/Dialect/TritonNvidiaGPU/IR/Ops.cpp

```diff
@@ -85,6 +85,15 @@ bool WarpGroupDotOp::needsPartialAccumulator() {
   return isFP8 && accFP32 && maxNumImpreciseAcc <= aTensorTy.getShape()[1];
 }
 
+bool WarpGroupDotOp::verifyDims() {
+  auto aShape = this->getA().getType().getShape();
+  auto bShape = this->getB().getType().getShape();
+
+  if (aShape[aShape.size() - 1] != bShape[aShape.size() - 2])
+    return false;
+  return true;
+}
```

Comment:
```suggestion
bool WarpGroupDotOp::verifyDims() {
  auto aShape = getA().getType().getShape();
  auto bShape = getB().getType().getShape();

  return aShape[aShape.size() - 1] == bShape[aShape.size() - 2];
}
```

same for the other checks